### PR TITLE
Add a token that's required to be invited.

### PR DESCRIPTION
**Pull request by lfn3**
Meant to be distributed in real life, or via email, or whatever other form you want. For those times when you don't want to invite everyone, and don't have a list of email addresses.

Totally understand if you don't want this, but I thought I'd offer it up.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -4,11 +4,12 @@ var request = require('request');
...
 
 router.get('/', function(req, res) {
-  res.render('index', { community: config.community });
+  res.render('index', { community: config.community,
+                        tokenRequired: config.inviteToken !== "" });
 });
 
 router.post('/invite', function(req, res) {
-  if (req.body.email) {
+  if (req.body.email && req.body.token && config.inviteToken !== "" && req.body.token === config.inviteToken) {
```
Comment (outsideris) [inline routes/index.js:12]: We should check the token only when `tokenRequired` is `ture` to use inviteToken as optional.

File: views/index.jade [hunk 2]
```diff
@@ -17,12 +17,15 @@ html
...
+                input(type="text", name="token", placeholder="Enter the token you were given")#slack-token.field
               input(type="submit", value="Join").submit
     script.
       var form = document.getElementById('join-form');
-      var email = document.getElementById('slack-email')
+      var email = document.getElementById('slack-email');
+      var token = document.getElementById('slack-token');
       form.addEventListener('submit', function(evt) {
-        if (!email.value) {
+        if (!email.value || (tokenRequired && !token.value)) {
```
Comment (outsideris) [inline views/index.jade:28]: Is it work? `tokenRequired` seems node.js variable not a variable in browser javascript.

**Comment by outsideris** [comment]
It can be useful. Can you add some description in README?

File: views/index.jade (hunk 2 above)
Comment (lfn3) [inline views/index.jade:28]: Yeah, tokenRequired is supplied to the jade template, but not available in node.

File: routes/index.js (hunk 1 above)
Comment (lfn3) [inline routes/index.js:12]: The token is checked when it's not set to an empty string, i.e. when someone has configured it. It is optional as a result.

**Comment by lfn3** [comment]
I'll add something to the readme now.

**Comment by outsideris** [comment]
@lfn3 Sorry, I was super busy. I tested it but there are a `tokenRequired is not defined` error. 
Because I think this is good feature and I am super late, I will merge it and fix it above your commit.

**Pull request merged**